x-pack/plugins/maps/public/classes/styles/vector/components/symbol/icon_stops.js

```diff
@@ -43,9 +43,22 @@ export function getFirstUnusedSymbol(iconStops) {
   return firstUnusedSymbol ? firstUnusedSymbol.value : DEFAULT_ICON;
 }
 
-export function IconStops({ field, getValueSuggestions, iconStops, onChange }) {
-  return iconStops.map(({ stop, icon }, index) => {
-    const onIconSelect = (selectedIconId) => {
+export function IconStops({
+  field,
+  getValueSuggestions,
+  iconStops,
+  onChange,
+  onCustomIconsChange,
+  customIcons,
+}) {
+  return iconStops.map(({ stop, icon, iconSource }, index) => {
+    const iconInfo =
+      iconSource === ICON_SOURCE.CUSTOM
+        ? customIcons.find(({ symbolId }) => symbolId === icon)
+        : getMakiSymbol(icon);
+    if (!iconInfo) return;
```

Comment:
This may cause problems. You are in map function so returning undefined will return an array of ReactNode and undefined. How about adding a filter after iconStops.map to remove undefineds?